util/stream.go

```diff
@@ -143,17 +151,21 @@ func (m *MessageStream) inbound() {
 
 // Parse incoming message
 func (m *MessageStream) parse() {
+	errMessage := "received: %v and encountered error: %v"
 	for {
-		b := <-m.pool.Full
-		log.Debugf("Rcvd: %v", b.Bytes())
-		msg, err := m.parser.Parse(b.Bytes())
-		// Log all message parsing errors.
-		if err != nil {
-			log.Print(err)
-		}
+		select {
```

Comment:
This select is needed to be able to shut down the goroutines. It has no performance impact. The goroutine which processes outgoing messages is also using a select and we haven't run into any issues because of that.

I've run the benchmarks and there was no significant difference between the old code and the new code.